DuggaSys/tests/showDuggaservice_test.php

```diff
@@ -0,0 +1,160 @@
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'opt' => 'UPDATEAU',
+            'hash' => 'tj7dh2nb',
+            'AUtoken' => '999'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'update active users' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'service' => 'localhost/LenaSYS/DuggaSys/showDuggaservice.php',
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'opt' => 'UPDATEAU',
+            'hash' => 'hjk4ert6',
+            'AUtoken' => '9999'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'get data from userAnswer' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
+        'service' => 'localhost/LenaSYS/DuggaSys/showDuggaservice.php',
+        'service-data' => serialize(array( // Data that service needs to execute function
+            'opt' => 'SAVDU',
+            'hash' => 'ghj1jfg2'
+        )),
+        'filter-output' => serialize(array( // Filter what output to use in assert test, use none to use all ouput from service
+            'none'
+        )),
+    ),
+    'update submitted dugga' => array(
+        'expected-output' => '{"debug":"[Guest] Missing hash\/password\/variant!","param":"{}","answer":"UNK","danswer":"UNK","score":0,"highscoremode":"","grade":"UNK","submitted":"","marked":"","deadline":"UNK","release":"UNK","files":[],"userfeedback":"UNK","feedbackquestion":"UNK","variant":"UNK","ishashindb":false,"variantsize":"UNK","variantvalue":"UNK","password":"UNK","hashvariant":"UNK","isFileSubmitted":"UNK","isTeacher":0,"variants":[],"duggaTitle":"UNK","hash":"UNK","hashpwd":"UNK","opt":"UNK","link":"UNK"}',
```

Comment:
a query-before-test is needed to be able to update a dugga. query-before-test INSERT INTO course(cid, creator) VALUES (9999, 1);

Reply:
anoither query-before-test is needed to insert listentries. INSERT INTO listentries(cid,entryname,link,kind,pos,creator,visible,vers,gradesystem,highscoremode,feedbackenabled,feedbackquestion) VALUES (9999,  "Inserttobedeleted", "UNK", 4, 12, 2, 1, 1337, 1, 1, 0, "UNK");

Reply:
after this you need another query-before-test SELECT MAX(lid) FROM listentries
This value needs to be saved in query-variables as lid.
lid has to be stated in the send as 'lid' => <!query-before-test3!><*[0][listenentries]*>

Reply:
then another query-before-test4 needs to be added(I've missed numbering all, but they start on 1 and increment for each new query, important to have these in the correct order.) query-before-test4 INSERT INTO userAnswer(cid,hash,password,moment) VALUES (9999, “dfg4zxc5”, “asdfasdf”, ?);

This questionmark is the saved value from query-variables.